### PR TITLE
Close subscription on destroy

### PR DESCRIPTION
**Pull request by maciaszczykm**

### CODE REVIEW

File: src/app/frontend/common/dialogs/previewdeployment/dialog.ts [hunk 1]
```diff
@@ -43,27 +45,20 @@ export class PreviewDeploymentDialog implements OnInit {
     const configuration = this.toDeploymentConfiguration(this.data.spec);
     this.text = JSON.stringify(this.removeEmptyField(configuration));
 
-    this.buttonToggleGroup.valueChange.subscribe((selectedMode: EditorMode) => {
+    this.buttonToggleGroup.valueChange.pipe(takeUntil(this.unsubscribe_)).subscribe((selectedMode: EditorMode) => {
```
Comment (maciaszczykm) [inline src/app/frontend/common/dialogs/previewdeployment/dialog.ts:48]: Most important part.

Comment (xiaoyang-sde) [reply src/app/frontend/common/dialogs/previewdeployment/dialog.ts:48]: It seems that `src/app/frontend/common/dialogs/editresource/dialog.ts` and `src/app/frontend/crd/crdobject/component.ts` have the same issue.

Comment (maciaszczykm) [reply src/app/frontend/common/dialogs/previewdeployment/dialog.ts:48]: Done.

**Comment by maciaszczykm** [comment]
@floreks PTAL

**Pull request merged**